src/common/reducers/repositories.js

```diff
@@ -1,42 +1,117 @@
+import merge from 'lodash/merge';
+import union from 'lodash/union';
+
 import * as ActionTypes from '../actions/repositories';
-import { parseGitHubRepoUrl } from '../helpers/github-url';
+
+// XXX move to entities.js
+// XXX should we be handling snaps here too?
+// no, because it's not part of the normalized result from github
+// so this should really be called github_entities
+// no, decompose instead ...
+export function entities(state = { snaps: {}, repos: {} }, action) {
+  if (action.payload && action.payload.entities) {
+    return merge({}, state, action.payload.entities);
+  }
+
+
+  const reposInitialState = {
+    __isSelected: false,
+    __isFetching: false,
+    __error: null
+  };
+
+  switch(action.type) {
+    case ActionTypes.REPOSITORY_TOGGLE_SELECT: {
+      const wasSelected = state.repos[action.payload.id].__isSelected;
+
+      return {
+        ...state,
+        repos: {
+          ...state.repos,
+          [action.payload.id]: {
+            ...state.repos[action.payload.id] || {},
+            ...reposInitialState, // clear state from building by deselecting
+            __isSelected: !wasSelected
+          }
+        }
+      };
```

Comment:
I think I've seen in redux examples that sometimes they would have a separate reducer for single entity (repo in our case), that will internally deal with state change for a single repo, and `entities` reducer will only need to select the entity based on id.

Mind my simplified pseudo-code:
```js
repo = (state, action) => {
  switch(action.type) {
  case REPOSITORY_TOGGLE_SELECT:
    return {
      ...state,
      __isSelected: !state.__isSelected
    }
  }
}

entities = (state, action) => {
switch(action.type) {
  case REPOSITORY_TOGGLE_SELECT:
    return {
      ...state,
      repos: {
        ...state.repos,
        [action.payload.id]: repo(state.repos[action.payload.id], action)
      }
    }
  }
}
```